### PR TITLE
JE implement buy operation

### LINKED ISSUE #14: FEATURE: Automate market price changes
### Automate Market Price Changes
Every day, the market price for cows should fluctuate within predefined bounds. Market price fluctuation should incentivize players to be greedy and purchase more cows when the price dips, perpetuating the tragedy of the commons.

**Requirements**

- [x] Admin should be able to define standard deviation for the price of cows to fluctuate.
- [x] Cow price should fluctuate daily, centered around an average that the admin will define.

**Stretch goals**
- [x]  Market price changes should realistically implement supply/demand

### PR DESCRIPTION
**Pull request by MrJacobEisner**
## MERGE je-priceChange-entity-repository BEFORE THIS
In this PR, we implemented the price increase on buy operation in userCommons. This calls a method in the commons which increases the cow price in the commons. This is related to issue https://github.com/ucsb-cs156-s23/proj-happycows-s23-6pm-3/issues/14

### CODE REVIEW

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java [hunk 1]
```diff
@@ -57,18 +57,25 @@ public class UserCommonsControllerTests extends ControllerTestCase {
...
   public void test_getUserCommonsById_exists_admin() throws Exception {
-  
+
+    Commons dummyCommons = dummyCommons(1);
+    when(commonsRepository.findById(1L)).thenReturn(Optional.of(dummyCommons));
     UserCommons expectedUserCommons = dummyUserCommons(1);
     when(userCommonsRepository.findByCommonsIdAndUserId(eq(1L),eq(1L))).thenReturn(Optional.of(expectedUserCommons));
 
-    MvcResult response = mockMvc.perform(get("/api/usercommons/?userId=1&commonsId=1"))
+    MvcResult response = mockMvc.perform(get("/api/usercommons?commonsId=1&userId=1"))
```
Comment (Biggergig) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:81]: any reason to swap these?

**Review by Biggergig — DISMISSED** [review]

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java (hunk 1 above)
Comment (awaang) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:81]: also may still need the forward slash after usercommons

**Review by awaang — CHANGES_REQUESTED** [review]

**Review by awaang — DISMISSED** [review]

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java (hunk 1 above)
Comment (MrJacobEisner) [reply src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:81]: I needed this to pass tests, the call is slightly different

**Review by pconrad — APPROVED** [review]
LGTM - will test on qa

**Review by pconrad — DISMISSED** [review]
LGTM - will test on qa

**Pull request closed without merging**